pyrene/src/components/DropdownButton/DropdownButton.tsx

```diff
@@ -1,96 +1,94 @@
-import React, { useState } from 'react';
-import PropTypes from 'prop-types';
+import React, { FunctionComponent, useState } from 'react';
 import clsx from 'clsx';
 import Popover from '../Popover/Popover';
 import OptionsList from './OptionsList';
-import './dropdownButton.css';
 import Loader from '../Loader/Loader';
+import './dropdownButton.css';
 
 /**
  * Dropdown Buttons are used primarily to group actions of a common topic.
  * They have the same appearance and behaviour as the regular Button.
  */
-const DropdownButton = (props) => {
 
-  const [state, setState] = useState({
-    displayActions: false,
-  });
+export interface DropdownButtonProps {
+  /**
+   * Array of action objects holding a Label, Icon and its associated Javascript event handler.
+   */
+  actions?: Array<{
+    label: string,
+    onClick: () => void,
+  }>,
+  /**
+   * Sets the alignment of the dropdown relative to the button.
+   */
+  align?: 'start' | 'end',
+  /**
+   * Disables any interaction with the component.
+   */
+  disabled?: boolean,
+  /**
+   * Adds an icon to the element.
+   */
+  icon?: string,
+  /**
+   * Sets the label displayed to the user.
+   */
+  label: string,
+  /**
+   * Disables the component and displays a loader inside of it.
+   */
+  loading?: boolean,
+}
 
-  const onClose = () => setState({ displayActions: false });
+const DropdownButton: FunctionComponent<DropdownButtonProps> = ({
+  actions = [{ label: 'tbd', onClick: () => null }],
```

Comment:
Seems like this was also in the JS implementation, but looks weird, doesn't it?

Reply:
> Seems like this was also in the JS implementation, but looks weird, doesn't it?

@neonnoon You are right. What we can do is to create a separate PR for enhancing that. This PR is just about translating from JS to TS.